### PR TITLE
ablog: init at 0.10.23

### PR DESCRIPTION
**Pull request by rgrinberg**
###### Description of changes

Introduce the package [ablog](https://ablog.readthedocs.io/en/latest/). Blogging software based on sphinx.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -14171,4 +14171,10 @@
     github = "nigelgbanks";
     githubId = 487373;
   };
+  rgrinberg = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:14174]: Please put this into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -34941,4 +34941,6 @@ with pkgs;
   honeytail = callPackage ../servers/tracing/honeycomb/honeytail { };
 
   honeyvent = callPackage ../servers/tracing/honeycomb/honeyvent { };
+
+  ablog = callPackage ../applications/misc/ablog { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:34945]: Please sort this as well.

File: pkgs/applications/misc/ablog/default.nix [hunk 3]
```diff
@@ -0,0 +1,37 @@
+{ lib
+, python3
+, python3Packages
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/ablog/default.nix:3]:
```suggestion
```

File: pkgs/applications/misc/ablog/default.nix [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+}:
+
+with python3Packages;
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/ablog/default.nix:6]:
```suggestion
with python3.pkgs;
```

File: pkgs/applications/misc/ablog/default.nix [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    invoke
+    watchdog
+    python-dateutil
+  ];
+
+  checkInputs = [
+    pytestCheckHook
+  ];
+
+  nativeBuildInputs = with python3.pkgs; [ setuptools-scm ];
```
Comment (SuperSandro2000) [inline pkgs/applications/misc/ablog/default.nix:29]:
```suggestion
  nativeBuildInputs = [ setuptools-scm ];
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: maintainers/maintainer-list.nix [hunk 6]
```diff
@@ -14171,4 +14177,10 @@
...
+    name = "Rudi Grinberg";
+    email = "me@rgrinberg.com";
+    github = "rgrinberg";
+    githubId = 139003;
+  };
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:14185]:
```suggestion
```

Comment (rgrinberg) [reply maintainers/maintainer-list.nix]: Fixed. Nice catch.
